destination/destination.go

```diff
@@ -99,16 +99,34 @@ func (d *Destination) Open(ctx context.Context) error {
 // Write routes incoming records to their appropriate handler based on the
 // operation.
 func (d *Destination) Write(ctx context.Context, recs []sdk.Record) (int, error) {
-	for i, r := range recs {
-		// TODO send all queries to postgres in one round-trip
-		err := sdk.Util.Destination.Route(ctx, r,
-			d.handleInsert,
-			d.handleUpdate,
-			d.handleDelete,
-			d.handleInsert,
-		)
+	b := &pgx.Batch{}
+	for _, rec := range recs {
+		var err error
+		switch rec.Operation {
+		case sdk.OperationCreate:
+			err = d.handleInsert(rec, b)
+		case sdk.OperationUpdate:
+			err = d.handleUpdate(rec, b)
+		case sdk.OperationDelete:
+			err = d.handleDelete(rec, b)
+		case sdk.OperationSnapshot:
+			err = d.handleInsert(rec, b)
+		default:
+			return 0, fmt.Errorf("invalid operation %q", rec.Operation)
+		}
 		if err != nil {
-			return i, err
+			return 0, err
+		}
+	}
+
+	br := d.conn.SendBatch(ctx, b)
+	defer br.Close()
+
+	for i := range recs {
+		_, err := br.Exec()
+		if err != nil {
+			// the batch is executed in a transaction, if one failed all failed
```

Comment:
maybe we should add this info to the error msg too? to show that more than one record failed to be written

Reply:
We return 0 so the SDK will nack all records in the batch, so the user will know that all failed. The error message will just point to the specific record that caused the batch to fail.